### DISCUSSION TITLE
How to add Custom Fonts With Existing Font List

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#6062] [Q&A]
<img width="558" alt="Screenshot 2024-08-16 at 8 21 55 AM" src="https://github.com/user-attachments/assets/b2ad782f-a0b8-484d-bd3f-f0584629f5cd">

It shows there is no method called AddOptions or SetOptions. Did I miss anything here?

**Comment by danstarns** [comment]
The type of your `customFont` var should be of type `Property` see - https://github.com/GrapesJS/grapesjs/blob/dev/src/style_manager/model/Property.ts

Please try `upValue` & `setValue` methods on here, if this doesn't work please report back with a reproducible example for us to help!

**Reply by RakulAgn** [reply]
SetValue and UpValue it's not Worked @danstarns for the reproducible example the setup is little harder on the do it could you please connect through any live section hope we can discuss about this and i took the reference from this code.

instead of adding fonts to the list by one by one i have set of custom fonts which already exist in database i am trying to load that fonts into my font-family property while editor on load method 

https://github.com/silexlabs/grapesjs-fonts/blob/main/src/fonts.js#L284

**Reply by RakulAgn** [reply]
Hey @danstarns Still Can't able to figure out the possibility of loading custom fonts the upValue and setValue is not working for my use-case

**Reply by artf** [reply]
The property has to be of `select` type, so if it's just about the TS error, you should be able to fix it in this way:
```js
import type { PropertySelect } from 'grapesjs';

(customFont as PropertySelect).addOption(...
```

**Reply by RakulAgn** [reply]
> The property has to be of `select` type, so if it's just about the TS error, you should be able to fix it in this way:
> 
> ```js
> import type { PropertySelect } from 'grapesjs';
> 
> (customFont as PropertySelect).addOption(...
> ```

I created a custom type for font-family so i can't access the propertySelect ??
